lucille-core/src/main/java/com/kmwllc/lucille/util/OpenSearchUtils.java

```diff
@@ -37,63 +37,61 @@ public class OpenSearchUtils {
    */
   public static OpenSearchClient getOpenSearchRestClient(Config config) {
     try {
-
       // get host uri
       URI hostUri = URI.create(getOpenSearchUrl(config));
-
       String userInfo = hostUri.getUserInfo();
 
       final var hosts = new HttpHost[]{
           new HttpHost(hostUri.getScheme(), hostUri.getHost(), hostUri.getPort())
       };
 
-
-      // code for Apache Client is taken from following link:
+      // code for building an Apache Client is inspired by the following link:
       // https://github.com/opensearch-project/opensearch-java/blob/main/samples/src/main/java/org/opensearch/client/samples/SampleClient.java
       // When comparing to example code, here are differences:
       //  - We gather data from our config rather than providing directly
       //  - We disable TLS/SSL verification only if acceptInvalidCerts is true (from config)
+
+      final BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
```

Comment:
Looks like a good (and straightforward) refactor to improve testability of this code. Since this is an error prone area of the codebase though -- even though we have unit tests that use mocking -- it would be ideal to also do a manual test to make sure we can still connect to a live opensearch instance... let's discuss.